Cicd actions build

Bodev13 opened the pull request.

Bodev13 committed:
Add GitHub Actions workflow and update README.md

Bodev13 committed:
Updated deployment.yaml file

Bodev13 committed:
update: removed GHCR steps and push of the images

Bodev13 committed:
Trigger deployment workflow

Bodev13 committed:
Trigger deployment workflow

Bodev13 committed:
changed git pull to git pull origin <my_branch>

Bodev13 committed:
Add GitHub Actions build deployment workflow

Bodev13 committed:
Add permissions for ghcr

Bodev13 committed:
updated deployment-gh-build.yaml for ghcr

Bodev13 committed:
test: trigger deployment

Bodev13 committed:
updated docker-compose to docker compose command in the deployment-gh-build.yaml

Bodev13 committed:
Trigger GH Actions deploy

Bodev13 committed:
Trigger GH Actions deploy

Bodev13 committed:
deleted old deployment.yaml file without ghcr

Bodev13 committed:
updated README.md for deployment part

Bodev13 committed:
added Deployment Checkliste.pdf

Bodev13 committed:
Trigger GH Actions deploy

Bodev13 committed:
rename deployment workflow to deployment.yaml. changed docker-compose to docker compose (plugin) in README.md

Bodev13 committed:
test deploy

Bodev13 committed:
test deploy

Bodev13 committed:
update depl.workflo and docker-compose for ghcr images and .env secrets

Bodev13 committed:
changed the path to the dockerfiles for BE and FE in deployment.yaml

Bodev13 committed:
updated paths to Dockerfiles for BE & FE in deployment.yaml

Bodev13 committed:
Test commit

Bodev13 committed:
Test commit

Bodev13 committed:
Test commit

Bodev13 committed:
Test commit

Bodev13 committed:
Test commit

Bodev13 committed:
test

Bodev13 committed:
fix: path to Dockerfiles

Bodev13 committed:
test path to Dockerfiles

Bodev13 committed:
test path to Dockerfiles

Bodev13 committed:
returned to a previous commit of the docker-compose.yml

Bodev13 committed:
updated deployment.yaml

Bodev13 committed:
reversed deployment.yaml back to working version

Bodev13 committed:
added main branch for push. changed actions/checkout to v5

Bodev13 committed:
changed actions/checkout to v6. added tags for BE and FE

Bodev13 committed:
added tags to BE & FE image

Bodev13 committed:
changes tags of BE & FE image to String format

Bodev13 committed:
added | to BE & FE tags

Bodev13 committed:
added .env variables for server

Bodev13 committed:
added update .env preparation

Bodev13 committed:
changed source for Upload Deployment Files

Bodev13 committed:
changed BE & FE in the docker-compose.yml from build to image...<SHA>

Bodev13 committed:
changed image in docker-compose from SHA to latest

Bodev13 committed:
Updated README CI/CD section. Updated deployment script in the deployment.yaml

mihaiandreineacsu commented on .github/workflows/deployment.yaml:
```diff
@@ -0,0 +1,105 @@
+            echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.repository_owner }} --password-stdin
+            docker compose pull
+            docker compose --env-file .env up -d --remove-orphans
+            docker image prune -f
```
@Bodev13 A quick rollback to the previous version is no longer possible.


```suggestion
            docker compose pull
            docker image prune -f
            docker compose --env-file .env up -d --remove-orphans
```

mihaiandreineacsu commented on README.md:
```diff
@@ -138,6 +139,52 @@ You can turn off and rebuild the container with the following commands
+
+Deployment happens automatically when you push changes to the branch `cicd-actions-build`
+
+You can also trigger the workflow manually from the `Actions tab` in GitHub.
```
@Bodev13 this is not really the case with your current `deployment.yml`

PascalNehlsen reviewed (changes requested):
Please focus on Mihai's review :) I missclicked!

Bodev13 committed:
Update .github/workflows/deployment.yaml

Suggested commit as a quick rollback to the previous version is no longer possible

Co-authored-by: mihaiandreineacsu <106650061+mihaiandreineacsu@users.noreply.github.com>

Bodev13 committed:
Update .github/workflows/deployment.yaml

Suggested change: added workflow_dispatch to the readme.md

Co-authored-by: mihaiandreineacsu <106650061+mihaiandreineacsu@users.noreply.github.com>

Bodev13 committed:
refactoring: Fix workflow_dispatch indentation
